salsa-2022-tests/tests/lru.rs

```diff
@@ -103,3 +118,74 @@ fn lru_doesnt_break_volatile_queries() {
         }
     }
 }
+
+#[test]
+fn lru_can_be_changed_at_runtime() {
+    let mut db = Database::default();
+    assert_eq!(load_n_potatoes(), 0);
+
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 32);
+
+    get_hot_potato::set_lru_capacity(&db, 64);
+    assert_eq!(load_n_potatoes(), 32);
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 64);
+
+    // Special case: setting capacity to zero disables LRU
+    get_hot_potato::set_lru_capacity(&db, 0);
+    assert_eq!(load_n_potatoes(), 64);
+    for i in 0..128u32 {
+        let input = MyInput::new(&mut db, i);
+        let p = get_hot_potato(&db, input);
+        assert_eq!(p.0, i)
+    }
+
+    // Create a new input to change the revision, and trigger the GC
+    MyInput::new(&mut db, 0);
+    assert_eq!(load_n_potatoes(), 192);
```

Comment:
Old salsa thinks it should be 128. https://github.com/salsa-rs/salsa/blob/80d0d141945051ec86ecd979ac06cca1e364b27e/tests/lru.rs#L119

Reply:
We did land some PRs fixing things about the LRU, most notably https://github.com/salsa-rs/salsa/pull/298, maybe that's related? Not obvious how. I have to file an issue to forward port those changes in particular.

Reply:
I just found out that 192 is because we created new inputs for `get_hot_potato`. Although the input value has not changed, the revision has changed and needs to be re-executed. We need to create all the inputs first, then here will be 128, logically equal to old salsa.